releasenotes/notes/0.19/refactor-grover-isgoodstate-check-54fdb61f899e5158.yaml

```diff
@@ -0,0 +1,29 @@
+features:
+  - |
+    When running the :class:`~qiskit.algorithms.Grover` algorithm class if the
+    optimal power is known and only a single circuit is run, the
+    :attr:`.AmplificationProblem.is_good_state` callback function is no longer
+    required to be set and the Grover search will return the most likely
+    bitstring.  Generally, if the optimal power of the Grover operator is not
+    known, the :class:`~qiskit.algorithms.Grover` algorithm checks different
+    powers (i.e.  iterations) and applies the
+    :attr:`~qiskit.algorithms.AmplificationProblem.is_good_state` function to
+    check whether a good bitstring has been measured.  For example, you are now
+    able to run something like::
+    
+    .. code-block::
+
+      from qiskit.algorithms import Grover, AmplificationProblem
+      from qiskit.providers.aer import AerSimulator
+      from qiskit.quantum_info import Statevector
+
+      # Fixed Grover power: 2.
+      grover = Grover(iterations=2, quantum_instance=AerSimulator())
+
+      # The ``is_good_state`` argument not required here since Grover search
+      # will be run only once, with a power of 2.
+      problem = AmplificationProblem(Statevector.from_label("111"))
+
+      # Run Grover search and print the best measurement
+      result = grover.amplify(problem)
+      print(result.top_measurement)  # should print 111
```

Comment:
This is wrongly rendered. 
<img width="606" alt="Screenshot 2021-12-03 at 15 49 20" src="https://user-images.githubusercontent.com/766693/144622617-22765119-f37c-4e2f-a4ae-1b2da326c289.png">



It might be the lack of trailing space?

```suggestion
      print(result.top_measurement)  # should print 111

```

Reply:
This is because I've accidentally used two different syntax forms for introducing a code block: the paragraph ends in `::`, but then it starts `.. code-block::` as well.  I think it may have been a merge mistake on my part - I think I updated the note locally, and then copied in some suggestion from Matthew or Abby without paying attention.

Reply:
Done in e624bd6.